src/poetry/inspection/lazy_wheel.py

```diff
@@ -614,7 +642,10 @@ def _extract_content_length(
 
             # This indicates that the requested range from the end was larger than the
             # actual file size: https://www.rfc-editor.org/rfc/rfc9110#status.416.
-            if code == codes.requested_range_not_satisfiable:
+            if (
+                code == codes.requested_range_not_satisfiable
+                and "Content-Range" in resp.headers
+            ):
                 # In this case, we don't have any file content yet, but we do know the
                 # size the file will be, so we can return that and exit here.
                 assert resp is not None
```

Comment:
this assertion is now too late to be useful: we've already used `resp` in the previous line

Reply:
I guess we can we can move it into the if condition instead.

Reply:
if `resp` was `None` then `code` would also be `None` so I guess the extra check really is unnecessary, I suppose I don't care either way